cmake/Modules/KatanaPythonSetupSubdirectory.cmake

```diff
@@ -214,7 +214,7 @@ function(add_python_setuptools_target TARGET_NAME)
       ALL
       COMMAND ${PYTHON_SETUP_COMMAND} ${quiet} build "$<$<CONFIG:Debug>:--debug>" ${parallel}
       COMMAND install ${PYTHON_ENV_SCRIPT}.tmp ${PYTHON_ENV_SCRIPT}
-      BYPRODUCTS ${PYTHON_BINARY_DIR} ${PYTHON_ENV_SCRIPT} ${CMAKE_BINARY_DIR}/katana_setup_requirements_cache.txt
+      BYPRODUCTS ${PYTHON_BINARY_DIR} ${PYTHON_ENV_SCRIPT} ${CMAKE_CURRENT_BINARY_DIR}/katana_setup_requirements_cache.txt
```

Comment:
If having the same byproduct for multiple targets is an issue for ninja, then maybe using a custom clean rule of some kind to delete this file would work. I only included it as a byproduct so that `make clean` would delete it.

Reply:
Looks like removing from katana_setup_requirements_cache.txt from BYPRODUCTS can fix the duplicated targets.

Does this file here really matters? Seems the first two is enough?

Reply:
If it's not listed here, it will not get removed by clean. So it does need to be listed somewhere to make sure it gets cleaned. I think cmake support auxillary clean targets. But I cannot remember how it works. I'm sure you can look it up.

Reply:
I've added "ADDITIONAL_CLEAN_FILES" in the updated commit.

Before this, ninja clean fails because remaining files. With ADDITIONAL_CLEAN_FILES, ninja clean works now.